client/state/lib/middleware.js

```diff
@@ -91,6 +95,22 @@ const updateSelectedSiteForDesktop = ( dispatch, action, getState ) => {
 	desktop.setSelectedSite( selectedSite );
 };
 
+/*
+ * Here be dragons.
+ */
```

Comment:
I was thinking I could make a `lib/sites-once` or something so that, in `state/lib/middleware`, we only have a call like `sitesOnce.addListener( … )` and a `sitesOnce.fireListeners`. What do you think?